include/onnxruntime/core/framework/ort_value.h

```diff
@@ -49,7 +49,6 @@ struct OrtValue {
   template <typename T>
   const T& Get() const {
     ORT_ENFORCE(onnxruntime::DataTypeImpl::GetType<T>() == type_, onnxruntime::DataTypeImpl::GetType<T>(), " != ", type_);
-    ORT_ENFORCE(IsAllocated(), "OrtValue contains no data");
```

Comment:
This call is not needed. In the kernels that support optional type, this call is performed before calling Get(). Even if the check is missed and it runs into a fatal issue (i.e.) the OrtValue has no data, we will know when we de-reference the nullptr in the next line. Seems unnecessary to pay the cost of this check twice.